staker-rewards/README.adoc

```diff
@@ -1,35 +1,108 @@
 = Staker rewards
 
-== Prerequisites
+ECDSA staker rewards are based on ETH locked and KEEP staked over time. The
+rewards calculations are performed weekly off-chain, and committed for
+distribution using a variant of Uniswap’s Merkle distributor This repository
+contains scripts used to calculate rewards and generate Merkle tree used by
+stakers to claim their rewards. Anyone can use tho scripts to validate reward
+distribution or to generate reward estimates for a reward interval still
+pending.
+
+== Algorithm
+
+The algorithm for calculating ECDSA staker rewards as implemented in the scripts 
+in this repository is described in this section.
+
+=== Input
+* Interval `T` starts at `T_start`.
+
+* Interval `T` ends at `T_end`.
+
+* All operators who have ever bonded ETH are taken into account.
+
+=== Operator Parameters
+For each Operator:
+
+* Calculate the amount of KEEP staked at `T_end` as `KEEP_staked`.
+
+* Calculate the total amount of ETH bonded at `T_start` as `ETH_bonded`.
+
+* Read the amount of ETH unbonded at `T_start` as `ETH_unbonded`.
+
+* Calculate the sum of ETH unbonded and withdrawn for the entire interval
+  as `ETH_withdrawn`.
+  
+* Calculate the total ETH under management as 
+  `ETH_total = ETH_bonded + ETH_unbonded - ETH_withdrawn`.
+
+=== Requirements
```

Comment:
It would be good to note what happens with the operator's allocation if requirements are not met.